algorithms/integration/processor.py

```diff
@@ -65,6 +65,73 @@
 logger = logging.getLogger(__name__)
 
 
+def assess_available_memory(params):
+
+    # Obtain information about system memory
+    available_memory = psutil.virtual_memory().available
+    available_swap = psutil.swap_memory().free
+    available_incl_swap = available_memory + available_swap
+    available_limit = available_incl_swap * params.block.max_memory_usage
+    available_immediate_limit = available_memory * params.block.max_memory_usage
+
+    # Compile a memory report
+    report = [
+        "Memory situation report:",
+    ]
+
+    def _report(description, value):
+        report.append("  %-50s:%5.1f GB" % (description, value))
+
+    _report("Available system memory (excluding swap)", available_memory / 1e9)
+    _report("Available swap memory", available_swap / 1e9)
+    _report("Available system memory (including swap)", available_incl_swap / 1e9)
+    _report(
+        "Maximum memory for processing (including swap)",
+        available_limit / 1e9,
+    )
+    _report(
+        "Maximum memory for processing (excluding swap)",
+        available_immediate_limit / 1e9,
+    )
```

Comment:
move the division into the helper function (I know who is to blame - I recognise my handwriting)
```suggestion
    def _report(description, numbytes):
        report.append("  %-50s:%5.1f GB" % (description, numbytes / 1e9))

    _report("Available system memory (excluding swap)", available_memory)
    _report("Available swap memory", available_swap)
    _report("Available system memory (including swap)", available_incl_swap)
    _report(
        "Maximum memory for processing (including swap)",
        available_limit
    )
    _report(
        "Maximum memory for processing (excluding swap)",
        available_immediate_limit
    )
```